File: lib/features/master_data/domain/entities/khach_hang.dart
// ============================================================================
// Domain Layer - Entities
// Based on UC_HKD_TT88_2021 - MD-05: Quản lý danh mục khách hàng
// ============================================================================

import 'package:equatable/equatable.dart';

class KhachHang extends Equatable {
  final String id;
  final String maKhachHang;
  final String tenKhachHang;
  final String? diaChi;
  final String? maSoThue;
  final String? soDienThoai;
  final String loaiKhachHang; // TO_CHUC / CA_NHAN / BAN_LE
  final String trangThai; // DANG_GIAO_DICH / NGUNG
  final String? createdAt;
  final String? updatedAt;

  const KhachHang({
    required this.id,
    required this.maKhachHang,
    required this.tenKhachHang,
    this.diaChi,
    this.maSoThue,
    this.soDienThoai,
    required this.loaiKhachHang,
    required this.trangThai,
    this.createdAt,
    this.updatedAt,
  });

  KhachHang copyWith({
    String? id,
    String? maKhachHang,
    String? tenKhachHang,
    String? diaChi,
    String? maSoThue,
    String? soDienThoai,
    String? loaiKhachHang,
    String? trangThai,
    String? createdAt,
    String? updatedAt,
  }) {
    return KhachHang(
      id: id ?? this.id,
      maKhachHang: maKhachHang ?? this.maKhachHang,
      tenKhachHang: tenKhachHang ?? this.tenKhachHang,
      diaChi: diaChi ?? this.diaChi,
      maSoThue: maSoThue ?? this.maSoThue,
      soDienThoai: soDienThoai ?? this.soDienThoai,
      loaiKhachHang: loaiKhachHang ?? this.loaiKhachHang,
      trangThai: trangThai ?? this.trangThai,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  List<Object?> get props => [
    id,
    maKhachHang,
    tenKhachHang,
    diaChi,
    maSoThue,
    soDienThoai,
    loaiKhachHang,
    trangThai,
    createdAt,
    updatedAt,
  ];
}

File: HkdAccounting.Domain/Entities/KhachHang.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HkdAccounting.Domain.Entities
{
    /// <summary>
    /// Khách hàng
    /// </summary>
    public class KhachHang
    {
        /// <summary>
        /// Khóa chính
        /// </summary>
        public long Id { get; set; }

        /// <summary>
        /// Mã khách hàng
        /// </summary>
        [Required(ErrorMessage = "Mã khách hàng là bắt buộc")]
        [MaxLength(20, ErrorMessage = "Mã khách hàng không được vượt quá 20 ký tự")]
        public string MaKhachHang { get; set; }

        /// <summary>
        /// Tên khách hàng
        /// </summary>
        [Required(ErrorMessage = "Tên khách hàng là bắt buộc")]
        [MaxLength(255, ErrorMessage = "Tên khách hàng không được vượt quá 255 ký tự")]
        public string TenKhachHang { get; set; }

        /// <summary>
        /// Mã số thuế
        /// </summary>
        [MaxLength(20, ErrorMessage = "Mã số thuế không được vượt quá 20 ký tự")]
        public string MaSoThue { get; set; }

        /// <summary>
        /// Địa chỉ
        /// </summary>
        [MaxLength(500, ErrorMessage = "Địa chỉ không được vượt quá 500 ký tự")]
        public string DiaChi { get; set; }

        /// <summary>
        /// Số điện thoại
        /// </summary>
        [MaxLength(20, ErrorMessage = "Số điện thoại không được vượt quá 20 ký tự")]
        public string SoDienThoai { get; set; }

        /// <summary>
        /// Email
        /// </summary>
        [MaxLength(100, ErrorMessage = "Email không được vượt quá 100 ký tự")]
        public string Email { get; set; }

        /// <summary>
        /// Khóa ngoại tới HkdInfo
        /// </summary>
        public long HkdInfoId { get; set; }

        /// <summary>
        /// Trạng thái
        /// </summary>
        [MaxLength(20, ErrorMessage = "Trạng thái không được vượt quá 20 ký tự")]
        public string TrangThai { get; set; } = "DANG_HOP_TAC";

        /// <summary>
        /// Ngày tạo bản ghi
        /// </summary>
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        /// <summary>
        /// Ngày cập nhật bản ghi
        /// </summary>
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public virtual HkdInfo HkdInfo { get; set; }
    }
}
